src/Executables/CombineH5/CombineH5.cpp

```diff
@@ -0,0 +1,144 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/program_options.hpp>
+#include <cstddef>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/SourceArchive.hpp"
+#include "IO/H5/VolumeData.hpp"
+#include "Parallel/Printf.hpp"
+#include "Utilities/FileSystem.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+// Compares source archives of each file against the 0th volume file. Returns
+// true if each source archive matches the 0th one
+bool check_src_files(const std::vector<std::string>& file_names) {
```

Comment:
maybe add `_match` at the end?

This also seems like something we could use elsewhere, so maybe add `src/IO/H5/CheckSourceFiles(Match).?pp`?

Reply:
I imagine something like this would never need to be its own executable so just `hpp` would be fine?

Reply:
Yes, sorry